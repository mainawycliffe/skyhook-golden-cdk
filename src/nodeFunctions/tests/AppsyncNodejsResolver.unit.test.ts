import "@aws-cdk/assert/jest";
import { Stack } from "@aws-cdk/core";
import { SynthUtils } from "@aws-cdk/assert";
import { GraphqlApi } from "@aws-cdk/aws-appsync";
import { AppsyncNodejsResolver } from "../..";

const testStack = new Stack();
const api = GraphqlApi.fromGraphqlApiAttributes(testStack, "api", {
  graphqlApiId: "testID",
});
new AppsyncNodejsResolver(testStack, "testFunction", {
  api,
  entry: "dummyEntry",
  fieldName: "booking",
  typeName: "Query",
});

it("creates a Lambda Function", () => {
  expect(testStack).toHaveResourceLike("AWS::Lambda::Function");
});

it("creates a data source for the lambda", () => {
  // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-datasource.html
  expect(testStack).toHaveResourceLike("AWS::AppSync::DataSource", {
    ApiId: "testID",
  });
});

it("creates a resolver with the data source", () => {
  // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-resolver.html
  expect(testStack).toHaveResourceLike("AWS::AppSync::Resolver", {
    ApiId: "testID",
    FieldName: "booking",
    TypeName: "Query",
  });
});

it("creates resources matching the snapshot", () => {
  const cfn = SynthUtils.toCloudFormation(testStack);
  expect(cfn).toMatchSnapshot();
});
