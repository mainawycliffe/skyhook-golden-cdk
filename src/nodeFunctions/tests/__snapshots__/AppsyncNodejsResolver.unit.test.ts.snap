// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates resources matching the snapshot 1`] = `
{
  "Resources": {
    "testFunctionDataSource4C03DC5E": {
      "Properties": {
        "ApiId": "testID",
        "Description": "Data source for field booking on type Query.",
        "LambdaConfig": {
          "LambdaFunctionArn": {
            "Fn::GetAtt": [
              "testFunctionLambda78269C17",
              "Arn",
            ],
          },
        },
        "Name": "testFunctionDataSource",
        "ServiceRoleArn": {
          "Fn::GetAtt": [
            "testFunctionDataSourceServiceRole2206622A",
            "Arn",
          ],
        },
        "Type": "AWS_LAMBDA",
      },
      "Type": "AWS::AppSync::DataSource",
    },
    "testFunctionDataSourceServiceRole2206622A": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "appsync.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "testFunctionDataSourceServiceRoleDefaultPolicy585B338D": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "testFunctionLambda78269C17",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "testFunctionLambda78269C17",
                          "Arn",
                        ],
                      },
                      ":*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "testFunctionDataSourceServiceRoleDefaultPolicy585B338D",
        "Roles": [
          {
            "Ref": "testFunctionDataSourceServiceRole2206622A",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "testFunctionLambda78269C17": {
      "DependsOn": [
        "testFunctionLambdaServiceRoleF9778236",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": "cdk-lambda-nodejs-mock-fake-bucket",
          "S3Key": "cdk-lambda-nodejs-mock-fake-lambda.ts",
        },
        "Description": "Appsync Resolver for field booking on type Query.",
        "MemorySize": 1024,
        "Role": {
          "Fn::GetAtt": [
            "testFunctionLambdaServiceRoleF9778236",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "testFunctionLambdaServiceRoleF9778236": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "testFunctionResolver84C7D9B2": {
      "DependsOn": [
        "testFunctionDataSource4C03DC5E",
      ],
      "Properties": {
        "ApiId": "testID",
        "DataSourceName": "testFunctionDataSource",
        "FieldName": "booking",
        "Kind": "UNIT",
        "TypeName": "Query",
      },
      "Type": "AWS::AppSync::Resolver",
    },
  },
}
`;
